querier.go

```diff
@@ -603,6 +603,10 @@ func (s *populatedChunkSeries) Next() bool {
 			continue
 		}
 
+		for _, chk := range s.chks {
```

Comment:
If we choose that implementing writing to the pool on queries is the best way to go, would there be any concern about doing this at the beginning of the Next() block? Then they can be used right away.

Reply:
yeah , I have been looking at this for the past 2 days and although it is an awkward place to put it couldn't find a better way to structure the code.